app/util/process_utils.py

```diff
@@ -48,3 +48,21 @@ def Popen_with_delayed_expansion(cmd, *args, **kwargs):
             cmd_with_deplayed_expansion.extend(cmd)
         cmd = cmd_with_deplayed_expansion
     return subprocess.Popen(cmd, *args, **kwargs)
+
+
+def get_environment_variable_setter_command(name, value):
+    """
+    Construct a platform specific command for setting an environment variable. Right now each command constructed
+    is designed to be chained with other commands.
+
+    :param name: The name of the environment variable
+    :type name: str
+    :param value: The value of the environment variable
+    :type value: str
+    :return: Platform specific command for setting the environment variable
+    :rtype: str
+    """
+    if os.name == 'nt':
+        return 'set {}={}&&'.format(name, value)
+    else:
+        return 'export {}="{}";'.format(name, value)
```

Comment:
I found this from https://docs.python.org/3.4/library/subprocess.html#subprocess.Popen:
"Note If specified, env must provide any variables required for the program to execute. On Windows, in order to run a side-by-side assembly the specified env must include a valid SystemRoot.
"

This is kind of interesting actually. E.g. In order to run nosetests, of course, you need to have `python` in `$PATH`. Right now, CR is operating with an assumption that the machine is already in a good state to run the job command, which is a fine assumption. If we want to go down the `env` route then we need to set all the environment variable explicitly, which right now are inherited from parent process. This actually can be beneficial if we make everything explicit (e.g. even install Python in setup if it's not in $PATH, etc.) But this is a breaking change so I doubt we are going to make this change right now.